sql/src/main/java/io/crate/action/job/ContextPreparer.java

```diff
@@ -626,6 +622,41 @@ public Boolean visitNestedLoopPhase(NestedLoopPhase phase, PreparerContext conte
             return true;
         }
 
+        @Override
+        public Boolean visitUnionPhase(UnionPhase unionPhase, PreparerContext context) {
+            RamAccountingContext ramAccountingContext = RamAccountingContext.forExecutionPhase(circuitBreaker, unionPhase);
+            ListenableRowReceiver downstreamRowReceiver = RowReceivers.listenableRowReceiver(
+                context.getRowReceiver(unionPhase, Paging.PAGE_SIZE));
+
+            FlatProjectorChain flatProjectorChain;
+            if (!unionPhase.projections().isEmpty()) {
+                flatProjectorChain = FlatProjectorChain.withAttachedDownstream(
+                    pageDownstreamFactory.projectorFactory(),
+                    ramAccountingContext,
+                    unionPhase.projections(),
+                    downstreamRowReceiver,
+                    unionPhase.jobId()
+                );
+            } else {
+                flatProjectorChain = FlatProjectorChain.withReceivers(Collections.singletonList(downstreamRowReceiver));
```

Comment:
this is senseless, can use downstreamRowReceiver directly as an argument to MultiUpstreamRowReceiver